tuf/utils/x509_test.go

```diff
@@ -239,6 +279,296 @@ func TestECDSAX509PublickeyID(t *testing.T) {
 	require.Equal(t, tufPrivKey.ID(), tufID)
 }
 
+func preserveEnv(name string) func() {
+	if env, has := os.LookupEnv(name); has {
+		os.Unsetenv(name)
+		return func() {
+			os.Setenv(name, env)
+		}
+	}
+
+	return func() {}
+}
+
+func TestExtractPrivateKeyAttributes(t *testing.T) {
+	if os.Getenv(t.Name()) != "1" {
```

Comment:
I agree with @riyazdf that this block, and the bit at the end where the test runs itself again, is a little confusing to read.

If I understand this thread correctly, it looks like the extra env checking is to make sure that we can run this test both with and without the FIPS flag on, whether or not it's set in environment?  

If my understanding is correct, would it be possible to factor this out so that the main body of the test itself is unexported, and the exported tests are:

`TestExtractPrivateKeyAttributesFIPSOn` and `TestExtractPrivateKeyAttributesNoFIPS` or something?

Each one can call the `preserveEnv` function, which will both save the previous environment variable and set it to a new desired value?

Similarly with the test below.